plugins/MauticOutlookBundle/Controller/PublicController.php

```diff
@@ -0,0 +1,143 @@
+            if (isset($query['d'], $query['sig'])) {
+                // get secret from Outlook plugin settings
+                $integrationHelper = $this->factory->getHelper('integration');
+                $outlookIntegration = $integrationHelper->getIntegrationObject('Outlook');
+                $keys = $outlookIntegration->getDecryptedApiKeys();
+
+                // generate signature
+                $salt = $keys['secret'];
+                if (strpos($salt, '$1$') === FALSE)
+                    $salt = '$1$'.$salt; // add MD5 prefix
+                $cr = crypt(urlencode($query['d']), $salt);
+                $mySig = hash('crc32b', $cr); // this hash type is used in c#
+
+                // compare signatures
+                if (hash_equals($mySig, $query['sig'])) {
+                    // decode and parse query variables
+                    $b64 = base64_decode($query['d']);
+                    $gz = gzdecode($b64);
+                    parse_str($gz, $query);
+                } else {
+                    // signatures don't match: stop
+                    unset($query);
+                }
+
+                if (!empty($query) && isset($query['email'], $query['subject'], $query['body'])) {
+
+                    /** @var \Mautic\EmailBundle\Model\EmailModel $model */
+                    $model = $this->getModel('email');
+
+                    // email is a semicolon delimited list of emails
+                    $emails = explode(';', $query['email']);
+                    $repo = $this->getModel('lead')->getRepository();
+
+                    foreach ($emails as $email) {
+                        $lead = $repo->getLeadByEmail($email);
+                        if ($lead === null) {
+                            $lead = $this->createLead($email, $repo);
+                        }
+
+                        if ($lead === null) continue; // lead was not created
+
+                        $idHash = hash('crc32', $email.$query['body']);
+                        $idHash = substr($idHash.$idHash, 0, 13); // 13 bytes length
+
+                        $stat = $model->getEmailStatus($idHash);
+
+                        // stat doesn't exist, create one
+                        if ($stat === null) {
+                            $this->addStat($lead, $email, $query, $idHash);
+                        } else { // Prevent marking the email as read on creation
+                            $model->hitEmail($idHash, $this->request); // add email event
+                        }
+                    }
+                }
+            }
+        }
+
+        return TrackingPixelHelper::getResponse($this->request); // send gif
+    }
+
+    /**
+     * @param $lead
+     * @param $email
+     * @param $query
+     * @param $idHash
+     */
+    public function addStat($lead, $email, $query, $idHash)
+    {
+        if ($lead !== null) {
+            $mailer = $this->factory->getMailer();
```

Comment:
Use `$this->get('mautic.helper.mailer')->getMailer()` here instead of `$this->factory->getMailer()`